Mathlib/RingTheory/PrincipalIdealDomain.lean

```diff
@@ -103,7 +103,18 @@ theorem mem_iff_eq_smul_generator (S : Submodule R M) [S.IsPrincipal] {x : M} :
 theorem eq_bot_iff_generator_eq_zero (S : Submodule R M) [S.IsPrincipal] :
     S = ⊥ ↔ generator S = 0 := by rw [← @span_singleton_eq_bot R M, span_singleton_generator]
 
-end Ring
+protected lemma FG {S : Submodule R M} (h : S.IsPrincipal) : S.FG :=
+  ⟨{h.generator}, by simp only [Finset.coe_singleton, span_singleton_generator]⟩
+
+instance _root_.PrincipalIdealRing.isNoetherianRing [IsPrincipalIdealRing R] :
```

Comment:
Please keep the lowered priority.

Reply:
Sure, but what effect does this have for prop valued instances? lean4 handles TC loops.

Reply:
The thing is if you have some `instance : Foo X`, then everytime the instance search sees a goal of type `Foo (some term)`,  this lemma will trigger, where usually we want the `Foo` to be derived from the structure of `some term`. See note [lower instance priority] for details.